### ISSUE TITLE
Setup hooks of _any_ sort

### ISSUE THREAD

**Issue by tarunik** [Zac-HD/hypofuzz#34]
Right now, Hypofuzz provides no hooks for when it starts running a test target.  This is a problem because the only time prior to an individual fuzz run one can do setup work is at import time, which makes it nearly impossible to _not_ do that work when running the testsuite conventionally (i.e. with pytest and Hypothesis, not Hypofuzz).

At the very least, a hook called from FuzzProcess.startup would let me do Hypofuzz-only setup steps instead of having to try to detect whether we're running under pytest or not during import, which may not work anyway since Hypofuzz delegates test collection (and importing) to pytest.

**Comment by tarunik** [comment]
Update: I was able to PoC this by manually monkeypatching FuzzProcess.startup, which'll get me going for now, but exposing a hook API of some sort would be appreciated.

**Comment by tarunik** [comment]
In my case, I was trying to inject an instance of my unit-under-test into the fuzz testcase.  (Normally, this would be done with a pytest fixture, but those don't play nice with hypofuzz.)

**Comment by Zac-HD** [comment]
Right, I think the right answer there is probably to support pytest fixtures, although I might leave function-scoped fixtures as an error.  That "just" takes setting them up and (for non-session-scoped) tearing them down when we switch between fuzz targets... it's conceptually pretty straightforward, just annoying to implement across various pytest versions.
